app/src/main/java/com/onelist/ui/MainActivity.kt

```diff
@@ -103,13 +103,14 @@ class MainActivity : ComponentActivity() {
     }
 
     @Composable
-    private fun ItemRow(name: String) {
+    private fun ItemRow(item: Item) {
```

Comment:
When you pass an object which is an instance of your dto you have access to every property and function as well. Strings have no way of knowing the other properties of the dto so you would need to write more lines of code to check if the values have changed, printing the properties together and declaring the objects before passing them as an argument. Firebase will also generate an id so passing the object will allow you to easily access that autogenerated value. https://kotlinlang.org/docs/classes.html#class-members.